### ISSUE TITLE
Fix Multi-Peer sharing

### ISSUE THREAD

**Issue by spersico** [statechannels/monorepo#1137]
**Describe the bug :bug:**
When more than 2 leechers try to download a torrent, the behaviour is kind of erratic, and most of the time, only one peer can leech at a time. This means that a big torrent first gets downloaded by one leecher, and then the other.
An error is shown in the console when this happens, that points to the problem being an `asyncToGenerator.js:6 uncaught promise error`

**Comment by geoknee** [comment]
This behaviour is much improved recently, and simultaneous leeching seems to mostly work well, at least with the fake provider. 

However, I still see an error when I have 2x leechers for the same file. If they run out of money and I wait a couple of minutes, I see messages indicating payments for unknown channels. I spent a bit of time digging around and found that we can sometimes end up with a `channelId={}`. This highlighted to me that our messages are of type `any`, and we could introduce some proper typing to protect against this kind of thing.

(`{}` is the default data on a message, if you pass undefined in to `paidStreamingExtension.stop()` as the `pseChannelId` parameter, `{}`s will get sent over the wire and interpreted as the `peerChannelId`)

This may or may not be related to wires being dropped #1215 .

TODO provide more context so we can reproduce this error.

**Comment by spersico** [comment]
So.. I was having, I think, some serious cache issues. Once I cleared all the `node_modules` and stuff, and pulled the latest changes, ran `yarn`, and started the project from the master branch, this error went away, I couldn't reproduce it.  Although a couple of hours ago this error did occur, around... 12-15 commits ago at the time of writing this. 

I tested a total of around 8 times, with 2 leechers and 1 seeder, with files of 100kb, 1mb, 3mb and 9mb, always using the FakeChannelProvider.
Although the 3mb and 9mb test files couldn't be downloaded completely by the leechers (because each peer has a budget to download a máximum of 2mb). Each peer was seeding and leeching correctly, and the download worked perfectly, as far as I could see. The leechers were sharing parts of the torrent between themselves

No errors of `asyncToGenerator`, nor the "wrong turn error, expected 3, index=1" that I saw a couple of days ago.

![image](https://user-images.githubusercontent.com/10502605/77017205-953c0a80-6958-11ea-82ea-410e35876636.png)

So.. 3 options, either this got fixed today (possible), I suck at testing (also possible), or we are really lucky, and this got magically fixed :grin:.

**Everything is not perfect though:**
The _NOWIRE_ still pops up, and for some reason, after the peers download the file, some Channel payment info starts disappearing (_the following screenshot is of the previous peer_):
![image](https://user-images.githubusercontent.com/10502605/77017461-578bb180-6959-11ea-82bf-a54a68039f51.png)

But @junjizhi's work will fix this up, I think.
